pilot/pkg/model/destination_rule.go

```diff
@@ -50,24 +50,33 @@ func (ps *PushContext) mergeDestinationRule(p *consolidatedDestRules, destRuleCo
 	}
 
 	if mdrList, exists := destRules[resolvedHost]; exists {
-		// `addRuleToProcessedDestRules` determines if the incoming destination rule would become a new unique entry in the processedDestRules list.
-		addRuleToProcessedDestRules := true
+		// `appendSeparately` determines if the incoming destination rule would become a new unique entry in the processedDestRules list.
+		appendSeparately := true
 		for _, mdr := range mdrList {
+			if exportToSet.Equals(mdr.exportTo) {
+				appendSeparately = false
+			} else if len(exportToSet.Difference(mdr.exportTo)) > 0 && len(mdr.exportTo.Difference(exportToSet)) > 0 {
```

Comment:
nit: can use SupersetOf, it doesn't compute a new Set that we immediately GC

Reply:
No ,we cannot use superset here. Say if a export to X, while b export export Y.